src/system_data.jl

```diff
@@ -1069,6 +1069,13 @@ end
 get_components(filter_func::Function, data::SystemData, attribute::SupplementalAttribute) =
     filter(filter_func, get_components(data, attribute))
 
+function get_components(data::SystemData, attribute_type::Type{<:SupplementalAttribute})
+    return [
+        get_component(data, x) for x in
+        list_associated_component_uuids(data.supplemental_attribute_manager, attribute_type)
+    ]
+end
```

Comment:
Should also define
```julia
get_components(filter_func::Function, data::SystemData, attribute_type::Type{<:SupplementalAttribute})
```
as we do for the `SupplementalAttribute` _instance_ one.

Reply:
(and then you should automatically get `get_available_components` from `ComponentContainer`)